discord/player.py

```diff
@@ -222,9 +246,25 @@ def _pipe_writer(self, source: io.BufferedIOBase) -> None:
                 self._process.terminate()
                 return
 
+    def _pipe_reader(self, dest: IO[bytes]) -> None:
+        while self._process:
+            if self._stderr is None:
+                return
+            try:
+                data: bytes = self._stderr.read(self.BLOCKSIZE)
+            except Exception:
+                _log.debug('Read error for %s, this is probably not a problem', self, exc_info=True)
+                return
+            if data is None:
+                return
+            # TODO: Do I need to check if this explodes?  catch OSError or something?
+            #       What happens if dest can no longer be written to?  Keep reading without writing or exit?
+            #       Does it pose an issue if the stderr pipe doesn't get read from?
+            dest.write(data)
```

Comment:
If an exception happens here then it should ideally properly clean up. It's probably as simple as just doing a `return` like you did elsewhere rather than propagating it to the thread exit.